utils.go

```diff
@@ -40,3 +40,35 @@ func validateMountOptions(mountOptions []string) error {
 	}
 	return nil
 }
+
+func applyNameOperation(oldNames []string, opParameters []string, op updateNameOperation) ([]string, error) {
```

Comment:
If all of the name manipulation logic ends up running through here, it's a good place to do the deduplication.